### PR TITLE
Rewrote CpsVmExecutorServiceTest to use ClassRule

### PR DESCRIPTION
**Pull request by jglick**
Made it a bit faster, and made it probably more legible than when using `ErrorCollector`.

#315 & #316 will need to be merged with this.

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorServiceTest.java [hunk 1]
```diff
@@ -27,167 +27,176 @@
...
-                p.setDefinition(new CpsFlowDefinition("echo(/see what ${-> 'this'} does/)", true));
-                WorkflowRun b = r.buildAndAssertSuccess(p);
-                r.assertLogContains(CpsVmExecutorService.mismatchMessage("WorkflowScript", "echo", "org.jenkinsci.plugins.workflow.cps.CpsClosure2", "call"), b);
-                r.assertLogNotContains("see what", b);
-                return null;
-            });
-            errors.checkSucceeds(() -> {
-                p.setDefinition(new CpsFlowDefinition("node {echo(/see what ${-> 'this'} does/)}", true));
-                WorkflowRun b = r.buildAndAssertSuccess(p);
-                r.assertLogContains(CpsVmExecutorService.mismatchMessage("WorkflowScript", "echo", "org.jenkinsci.plugins.workflow.cps.CpsClosure2", "call"), b);
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorServiceTest.java]: Largely reindentation, so hide WS pages in GH diff tab to review.

**Comment by sparshev** [comment]
Any updates guys?

**Comment by sparshev** [comment]
@jglick hopefully it's not stale because of my change was merged...

**Comment by jglick** [comment]
Yeah I need to resolve the merge conflict here.

File: src/test/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorServiceTest.java [hunk 2]
```diff
@@ -27,185 +27,181 @@
...
-            CpsVmExecutorService.FAIL_ON_MISMATCH = origFailOnMismatch;
-        }
+        p.setDefinition(new CpsFlowDefinition("org.jenkinsci.plugins.workflow.cps.CpsThread.current().getExecution().getShell().evaluate('echo \"Goofy\"', 'X.groovy')", false));
+        WorkflowRun b = r.buildAndAssertSuccess(p);
+        r.assertLogContains("Goofy", b);
+        r.assertLogNotContains("expected to call", b);
     }
 
     @Issue("JENKINS-58407")
-    @Ignore
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorServiceTest.java]: Seems to be passing.

**Comment by sparshev** [comment]
Yep, looks great)

**Comment by dwnusbaum** [comment]
Closing and reopening to get a CI build.

**Comment by omehegan** [comment]
A CloudBees customer has asked when the fix for #315 will be released. Is there any ETA that I can provide them with?

**Pull request merged**